pkg/reporegistry/reporegistry.go

```diff
@@ -32,6 +35,15 @@ func NewTestedDefault() (*RepoRegistry, error) {
 	return New(testReposPath)
 }
 
+// NewTestedDefaultUnitTests returns a new RepoRegistry instance with the data
+// loaded from the default test repositories, but for unit tests. The difference
+// is that the path to the repositories is relative to the directory of this
+// source file and not to the project root.
```

Comment:
Could this maybe be unified with something like:
```go
--- a/pkg/reporegistry/reporegistry.go
+++ b/pkg/reporegistry/reporegistry.go
@@ -2,6 +2,8 @@ package reporegistry
 
 import (
        "fmt"
+       "path/filepath"
+       "runtime"
 
        "github.com/osbuild/images/pkg/distroidparser"
        "github.com/osbuild/images/pkg/rpmmd"
@@ -28,7 +30,8 @@ func New(repoConfigPaths []string) (*RepoRegistry, error) {
 // NewTestedDefault returns a new RepoRegistry instance with the data
 // loaded from the default test repositories
 func NewTestedDefault() (*RepoRegistry, error) {
-       testReposPath := []string{"./test/data"}
+       _, srcfile, _, _ := runtime.Caller(0)
+       testReposPath := []string{filepath.Join(filepath.Dir(srcfile), "../../test/data")}
        return New(testReposPath)
 }
 
```
i.e. regardless how NewTestedDefault() we just always look relative to the `pkg/reporegistry/reporegistry.go` path? Or am I missing something here?

Reply:
Thanks for the suggestion, I didn't really put too much thought into this. :innocent: